cpp/CMakeLists.txt

```diff
@@ -177,6 +175,29 @@ SECTIONS
 ]=])
 endif()
 
+##############################################################################
+# - NVTX support in raft -----------------------------------------------------
+
+if (NVTX)
+  # This enables NVTX within the project with no option to disable it downstream.
+  target_link_libraries(raft INTERFACE CUDA::nvToolsExt)
+  target_compile_definitions(raft INTERFACE NVTX_ENABLED)
+  set(nvtx_export_string "")
+else()
+  # Allow enable NVTX downstream if not set here.
+  # This creates a new option at build/install time, which is set by default to OFF,
+  # but can be enabled in the dependent project.
+  get_property(nvtx_option_help_string CACHE NVTX PROPERTY HELPSTRING)
+  string(CONCAT nvtx_export_string
+  "option(NVTX \"" ${nvtx_option_help_string} "\" OFF)"
```

Comment:
This option should be namespaced ( `RAFT_NVTX` ) as right now it has a very high chance of clashing with an existing variable. By namespacing we allow finer grained control ( my project has NVTX, but raft wont )

Reply:
Thanks, for the review!